packages/engine/Source/Shaders/AtmosphereCommon.glsl

```diff
@@ -50,14 +48,46 @@ void computeScattering(
         return;
     }
 
+    // To deal with smaller values of PRIMARY_STEPS (e.g. 4)
+    // we implement a split strategy: sky or horizon.
+    //
+    // For performance reasons, instead of a if/else branch
+    // a soft choice is implemented through a weight 0.0 <= w_stop_gt_lprl <= 1.0
+    float lprl = length(primaryRayLength);
+    float x = 1e-7 * primaryRayAtmosphereIntersect.stop / lprl;
+    // w_stop_gt_lprl: similar to (1+tanh)/2
+    // Value close to 0.0: close to the horizon
+    // Value close to 1.0: above in the sky
+    float w_stop_gt_lprl = max(0.0,min(1.0,(1.0 + x * ( 27.0 + x * x ) / ( 27.0 + 9.0 * x * x ))/2.0));
```

Comment:
Can the `tanh` approximation be pulled out into a separate function? Something like:
```glsl
/**
 * Rational approximation to tanh(x)
*/
float approximateTanh(float x) {
    return x * (27.0 + x * x) / (27.0 + 9.0 * x * x);
}
``` 

I think it would make some of these long lines a little easier to read.

Reply:
Yes. I carefuly tested the performance implications, tried various options, and ended up with the one in https://github.com/CesiumGS/cesium/pull/11109/commits/2290ec24291d9d977492cf2fedbf24e68d81120d  : 
```c
/**
 * Rational approximation to tanh(x)
*/
float approximateTanh(float x) {
    float x2 = x * x;
    return max(-1.0, min(+1.0, x * (27.0 + x2) / (27.0 + 9.0 * x2)));
}
```